crates/runtime/src/model/tool_use.rs

```diff
@@ -303,6 +318,7 @@ impl ToolUsingChat {
     async fn chat_stream_inner(
         &self,
         req: CreateChatCompletionRequest,
+        is_root_call: bool,
```

Comment:
As discussed, let's use a wrapping stream that emits the metric once the stream finishes.